module/input/SpeechIntent/src/SpeechIntent.cpp

```diff
@@ -0,0 +1,405 @@
+                                                         __FILE__,
+                                                         __LINE__,
+                                                         errno));
+                return;
+            }
+            buffer[bytes_read] = 0;
+            NUClear::log<NUClear::FATAL>(fmt::format("({}:{})\nVOICE2JSON stderr, errno = {} ({} bytes)\n{}",
+                                                         __FILE__,
+                                                         __LINE__,
+                                                         errno,
+                                                         bytes_read,
+                                                         buffer));
+        });
+
+
+        on<Trigger<SpeechInputSetOutputMsg>>().then(
+            [this](const SpeechInputSetOutputMsg& msg) { output_enabled = msg.enabled; });
+
+        on<Trigger<SpeechInputRecognizeWavFile>>().then([this](const SpeechInputRecognizeWavFile& msg) {
+            if (!write_audio_to_file(voice2json_proc.stdin, (char*) msg.filename.data())) {
+                NUClear::log<NUClear::FATAL>(
+                    fmt::format("({}:{}) Failed to write to {}", __FILE__, __LINE__, msg.filename.data()));
+            }
+        });
+    }
+
+    void SpeechIntent::recognize_wav(std::string wav_filename) {
+        std::unique_ptr<SpeechInputRecognizeWavFile> msg =
+            std::make_unique<SpeechInputRecognizeWavFile>(SpeechInputRecognizeWavFile{wav_filename});
+        emit(std::move(msg));
+    }
+
+    SpeechIntent::SpeechIntent(std::unique_ptr<NUClear::Environment> environment)
+        : Reactor(std::move(environment)), config{} {
+
+
+        on<Configuration>("SpeechIntent.yaml").then([this](const Configuration& cfg) {
+            this->log_level = cfg["log_level"].as<NUClear::LogLevel>();
+        });
+
+        on<Trigger<CommandLineArguments>>().then([this](const CommandLineArguments& args) {
+            if (args.size() > 1 && args[1] == "file") {
+                if (args.size() != 3) {
+                    NUClear::log<NUClear::FATAL>(
+                        fmt::format("invalid number of arguments, expected 3, got {}", args.size()));
+                }
+                this->config.transcribe_mode = TRANSCRIBE_MODE_FILE;
+                std::string wav_filename     = std::string(args[2]);
+                log(wav_filename);
+
+                init();
+
+                on<Trigger<SpeechIntentMsg>>().then([this](const SpeechIntentMsg& msg) { print_intent(msg); });
+                recognize_wav(wav_filename);
+            }
+            else if (args.size() > 1 && args[1] == "cli") {
+                this->config.transcribe_mode = TRANSCRIBE_MODE_FILE;
+                init();
+
+                on<Trigger<SpeechIntentMsg>>().then([this](const SpeechIntentMsg& msg) { print_intent(msg); });
+                on<IO>(STDIN_FILENO, IO::READ).then([this] {
```

Comment:
Same with this, maybe some comments would clear things up, someone else might have a different opinion, but enabling this reaction if the specific use case for it is needed seems like a cleaner way to have this optionally included